vllm/engine/llm_engine.py

```diff
@@ -91,6 +91,10 @@ class OutputData(NamedTuple):
     is_async: bool
     is_last_step: bool
     skip: List[int]
+    # The step that this output corresponds to in multi-step.
+    # step_num is invalid when `outputs` have outputs from multiple
+    # steps.
+    step_num: Optional[int]
```

Comment:
Let's try to make a better name for this. Something like `step_idx_in_multi_step`? (I'm really bad at naming tho...)

Also if we want to introduce this attribute, can we remove `is_last_step`,  so that we don't need to add one more argument to most existing functions?

Reply:
`step_idx_in_multi_step` sounds good to me. 

I thought `step_idx` would be sufficient as when multi-step is disabled, it could be just be 0. But I see SeqGroupMetadata.state.step_num is only incremented when multi_step is enabled and it gets in the way of some logic (for example https://github.com/vllm-project/vllm/blob/2940afa04e39fa9f248c565687d9a2acf7401355/vllm/engine/llm_engine.py#L1244) - I think it can be generalized. 

about `is_last_step`, I think we should still keep it as we don't have "total number of steps" information to derive `is_last_step`.

Reply:
renamed to `step_idx_in_multi_step`